Site settings: Refactor to use locale rather than lang id

yoavf opened the pull request:
Lang ids depend on a hard coded config array, with meaningless values. We should use locale codes instead.

gwwar commented on client/lib/wpcom-undocumented/lib/undocumented.js:
```diff
@@ -329,10 +329,14 @@ Undocumented.prototype.settings = function( siteId, method, data, fn ) {
-		path: '/sites/' + siteId + '/settings',
-		body: data
-	}, fn );
+	var path = '/sites/' + siteId + '/settings',
```
I'm relatively sure we can use es6 in this file, so we can update this to:
```javascript
const path = `/sites/${ siteId }/settings`;
const params = { apiVersion: '1.2' }
```

gwwar commented:
Found a regression:
1. Select a site with no language set
2. Click on settings
Expected: We navigate to settings
Actual: We're redirected to stats

yoavf commented:
@gwwar thanks for the review, and sorry it too me so long to get back to this.
I've rebased and updated the syntax as recommended - thanks!

> Found a regression:
> 1. Select a site with no language set
> 2. Click on settings
> Expected: We navigate to settings
> Actual: We're redirected to stats

I couldn't reproduce, but maybe because of the 'site with no language set' - there might be some, but we currently prevent the option from being null or having a non valid value. Do you recall which site made this happen?

gwwar commented:
> but we currently prevent the option from being null or having a non valid value

If I remember correctly, previously the Calypso and wp-admin language options were out of sync. You could for example set en-gb in wp-admin, and when viewing language options in Calypso see nothing set. So the regression was probably fixed via the other issue.

gwwar commented:
@yoavf took this for a spin and it looks good to me 👍

gwwar commented:
As an aside, @retrofox any reason why we're using undocumented vs wpcomjs here for site settings?

retrofox commented:
> As an aside, @retrofox any reason why we're using undocumented vs wpcomjs here for site settings?

No reason. In fact we should try to leave to use `wpcom-undocumented` specially if the features are already implemented in `wpcom.js`.

```es6
import wp from 'lib/wp';

// create site settings instance
const siteSettings = wpcom.site( 'your-testing-blog').settings();

// get current site settings object
siteSettings.get()
  .then( data => console.log( data ) )
  .catch( cach => console.log( catch ) );

// get en specific option
siteSettings.getOption( 'language' )
  .then( option => console.log( option ) )
  .catch( cach => console.log( catch ) );

// update site settings
siteSettings.update( { title: 'Let\'s change the title', 'gtm_offset': 10 } )
  .then( data => console.log( data ) )
  .catch( catch => console.log( catch ) );

// update only one option
siteSettings.updateOption( title 'Let\'s change the title' )
  .then( data => console.log( data ) )
  .catch( catch => console.log( catch ) );
```

mtias commented:
@yoavf is this good to go?

yoavf commented:
Should be, sorry I let it slip. It's been a while so I'll test it again next week and merge

mtias commented:
No worries :)
